irods/test/connection_test.py

```diff
@@ -159,8 +160,18 @@ def test_assigning_session_connection_timeout_to_invalid_values__issue_569(self)
             with self.assertRaises(ValueError):
                 sess.connection_timeout = value
 
-        # Test previously set value is unaffected
-        self.assertEqual(sess.connection_timeout, DESIRED_TIMEOUT)
+    def test_assigning_session_connection_timeout_to_large_values__issue_623(self):
+        # Test use of a too-large timeout in iRODSSession constructor as well as on assignment to the
+        # iRODSSession property 'connection_timeout'.  In both cases, error checking and hard-limiting
+        # should be immediate.
+        sess = helpers.make_session(connection_timeout = MAXIMUM_CONNECTION_TIMEOUT + 1)
+        self.assertEqual(sess.connection_timeout, MAXIMUM_CONNECTION_TIMEOUT)
+
+        sess.connection_timeout = MAXIMUM_CONNECTION_TIMEOUT + 1
+        self.assertEqual(sess.connection_timeout, MAXIMUM_CONNECTION_TIMEOUT)
```

Comment:
Some minor paranoia here...

Perhaps we should set the timeout to some good value in between these two assertions in order to see that the setter took effect. Or, we could split the test up into one for the `__init__` function (i.e. `make_session`) and one for the setter.

The reason I am suggesting this is that we don't know that the `connection_timeout` is `MAXIMUM_CONNECTION_TIMEOUT` here because the setter used a too-large value. It could just be unchanged from when the `make_session` call used a too-large value. Not that there is a difference in this case...

Hopefully that makes sense.

Reply:
Yes, I see, I think... asserting the sess.connection_timeout value is really asserting the pool's value.  That is certainly not what is intended.

Reply:
Well... I actually was thinking about the fact that we are using the setter to set the value of `connection_timeout` and asserting that it then has the value that it already holds. The reason I have a problem with this is because the setter may just be leaving the value the way it was.

From this test, we cannot tell what it did. Did it set `connection_timeout` to some maximum value (as we are doing), or did it leave the value unchanged? That's why I'm suggesting setting it to some other, valid value in between so that we can see that the value actually was changed by the setter.

Something like this was what I had in mind:
```python
        # This sets the original value using the __init__ method - should be the "max" value.
        sess = helpers.make_session(connection_timeout = MAXIMUM_CONNECTION_TIMEOUT + 1)
        self.assertEqual(sess.connection_timeout, MAXIMUM_CONNECTION_TIMEOUT)
        # Then we set the value to something different...
        sess.connection_timeout = 1 # (or some other valid value)
        self.assertEqual(sess.connection_timeout, 1)
        # Now we set the invalid value, asserting that the setter is using the "max" value
        # and this ensures that it's not just unchanged from its previous value.
        sess.connection_timeout = MAXIMUM_CONNECTION_TIMEOUT + 1
        self.assertEqual(sess.connection_timeout, MAXIMUM_CONNECTION_TIMEOUT)
```

Reply:
Ok think I've done it.... @alanking , please examine

Reply:
Yes, I think so. I've left a comment on the change, so I'll resolve this one as the major concern has been addressed. Thanks!